code/model/editableformfields/EditableFormField.php

```diff
@@ -365,6 +365,13 @@ public function canEdit($member = null) {
         $parent = $this->Parent();
 		if($parent && $parent->exists()) {
 			return $parent->canEdit($member) && !$this->isReadonly();
+		} else if ($this->ID == 0) {
+			// This is for GridFieldOrderableRows support as it checks edit permissions on 
+			// singleton of the class. Fixes backwards compatibility issues where 
+			// 'Content Authors' used to be able to re-order items on the old interface 2.x version.
+			$parentClass = $this->config()->has_one['Parent'];
+			$parent = singleton($parentClass);
+			return $parent->canEdit($member);
```

Comment:
Look at how GridFieldOrderableRows works:
https://github.com/silverstripe-australia/silverstripe-gridfieldextensions/blob/master/code/GridFieldOrderableRows.php

It uses singleton($className)->canEdit() to check if you can reorder. Switching this on the GridFieldOrderableRows side to check each ID with the Classname would slow down the snappyness of the search, so I dont feel like that's an option.

Alteratively, Editable Form Fields -could- just be updated to do a array("ADMIN", "SITETREE_EDIT_ALL") permission check like on SiteTree::canEdit().